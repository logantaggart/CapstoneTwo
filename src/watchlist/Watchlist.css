.list {
    text-align: center;
    margin: 100px;
    color: seashell
}

.info {
    margin-left: 175px;
    margin-right: 175px
}