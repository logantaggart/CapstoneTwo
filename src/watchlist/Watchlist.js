import React, { useState, useContext } from 'react'

import UserContext from '../auth/UserContext'

import './Watchlist.css'

function Watchlist(props) {
    const [formData, setFormData] = useState({
        sortBy: "name",
        orderBy: ""
    })

    const { currentUser, watchlist } = useContext(UserContext)

    const is_testing = false
    const test_watchlist = {
        "watchlist":
            [{
                "currency": "USD",
                "date": "2021-06-03T02:05:49.000Z",
                "market_cap": 698208752460.92,
                "name": "Bitcoin",
                "price": 37287.18,
                "volume": 32850986431.73,
                "username": 'ciewicnjqwicnuehfu3fbwkrenfujn3'
            },
            {
                "currency": "USD",
                "date": "2021-06-03T00:39:10.000Z",
                "market_cap": 55504648942.88,
                "name": "Cardano",
                "price": 1.7373,
                "volume": 3305794201.01,
                "username": 'ciewicnjqwicnuehfu3fbwkrenfujn3'
            }]
    }



    let data = !is_testing ? watchlist['watchlist'] : test_watchlist['watchlist']

    function handleSubmit(evt) {
        evt.preventDefault()

        props.retrieveWatchlist(currentUser.username, formData.sortBy, formData.orderBy)
    }
    function handleChange(evt) {
        const { name, value } = evt.target
        setFormData(s => ({ ...s, [name]: value }))
    }

    return (
        <div className="list">
            <h3>Cryptocurrency Watchlist:</h3>
            <br />

            <form onSubmit={handleSubmit} className="container">
                <select className="form-control" name="sortBy" onChange={handleChange} value={formData.sortBy}>
                    <option value="name">Alphabetical</option>
                    <option value="price">Price</option>
                    <option value="market_cap">Market Cap</option>
                    <option value="volume">Volume</option>
                    <option value="date">Date</option>
                </select>
                <br />

                <select className="form-control" name="orderBy" onChange={handleChange} value={formData.orderBy}>
                    <option value="ASC">Ascending</option>
                    <option value="DESC">Descending</option>
                </select>
                <br />

                <button className="btn btn-md btn-primary">Sort</button>
            </form>
            <br />
            <hr />

            <div className="info">
                {data.map((data, idx) => (
                    <div key={idx}>
                        <h5>{data.name}</h5>
                        <b>Price:</b> <u>{data.price} {data.currency}</u>
                        <br />
                        <b>Market Cap:</b> <u>{data.market_cap} {data.currency}</u>
                        <br />
                        <b>Volume:</b> <u>{data.volume} {data.currency}</u>
                        <br /> <br />
                        <i>Information Accurate As Of: {data.date.split("T")[0]}</i>
                        <br /> <br />
                        <hr />
                    </div>
                ))}
            </div>
        </div>
    )
}

export default Watchlist