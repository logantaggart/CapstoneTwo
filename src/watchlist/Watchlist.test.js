import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import UserProvider from '../testUtils'

import Watchlist from './Watchlist'

it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <Watchlist />
            </UserProvider>
        </MemoryRouter>
    )
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <UserProvider>
            <Watchlist />
        </UserProvider>
    )

    expect(asFragment()).toMatchSnapshot()
})