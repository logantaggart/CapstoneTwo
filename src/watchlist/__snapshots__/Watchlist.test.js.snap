// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <h3>
      Cryptocurrency Watchlist:
    </h3>
    <br />
    <form
      class="container"
    >
      <select
        class="form-control"
        name="sortBy"
      >
        <option
          value="name"
        >
          Alphabetical
        </option>
        <option
          value="price"
        >
          Price
        </option>
        <option
          value="market_cap"
        >
          Market Cap
        </option>
        <option
          value="volume"
        >
          Volume
        </option>
        <option
          value="date"
        >
          Date
        </option>
      </select>
      <br />
      <select
        class="form-control"
        name="orderBy"
      >
        <option
          value="ASC"
        >
          Ascending
        </option>
        <option
          value="DESC"
        >
          Descending
        </option>
      </select>
      <br />
      <button
        class="btn btn-md btn-primary"
      >
        Sort
      </button>
    </form>
    <br />
    <hr />
    <div
      class="info"
    >
      <div>
        <h5>
          Bitcoin
        </h5>
        <b>
          Price:
        </b>
         
        <u>
          37287.18 USD
        </u>
        <br />
        <b>
          Market Cap:
        </b>
         
        <u>
          698208752460.92 USD
        </u>
        <br />
        <b>
          Volume:
        </b>
         
        <u>
          32850986431.73 USD
        </u>
        <br />
         
        <br />
        <i>
          Information Accurate As Of: 2021-06-03
        </i>
        <br />
         
        <br />
        <hr />
      </div>
      <div>
        <h5>
          Cardano
        </h5>
        <b>
          Price:
        </b>
         
        <u>
          1.7373 USD
        </u>
        <br />
        <b>
          Market Cap:
        </b>
         
        <u>
          55504648942.88 USD
        </u>
        <br />
        <b>
          Volume:
        </b>
         
        <u>
          3305794201.01 USD
        </u>
        <br />
         
        <br />
        <i>
          Information Accurate As Of: 2021-06-03
        </i>
        <br />
         
        <br />
        <hr />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
